torch/testing/_internal/common_methods_invocations.py

```diff
@@ -8268,15 +8283,25 @@ def __call__(self, opinfo, device, dtype, requires_grad, **kwargs):
         assert isinstance(num_input_tensors, list)
         _foreach_inputs_kwargs = {k: kwargs.pop(k, v) for k, v in _foreach_inputs_default_kwargs.items()}
 
-        for num_tensors in num_input_tensors:
-            for rightmost_arg_type in self._rightmost_arg_types:
-                input = sample_inputs_foreach(None, device, dtype, num_tensors, **_foreach_inputs_kwargs)
+        _size0_tensors_test_done = False
+        for num_tensors_, rightmost_arg_type in itertools.product(num_input_tensors, self._rightmost_arg_types):
+            for zero_size in (False, True) if "cuda" in device and dtype == torch.float32 else (False,):
```

Comment:
instead of making additional nest level, maybe just spell out zero-size test outside of this loop